### PR TITLE
IGNITE-12024: [ML] PeerClassloading for ml related lambdas

### PR DESCRIPTION
**Pull request by avplatonov**
@zaleslaw, Please review this PR

### CODE REVIEW

File: modules/ml/src/main/java/org/apache/ignite/ml/composition/ModelsComposition.java [hunk 1]
```diff
@@ -23,13 +23,15 @@
...
 import org.apache.ignite.ml.composition.predictionsaggregator.PredictionsAggregator;
+import org.apache.ignite.ml.environment.deploy.DeployableObject;
 import org.apache.ignite.ml.math.primitives.vector.Vector;
 import org.apache.ignite.ml.util.ModelTrace;
 
 /**
  * Model consisting of several models and prediction aggregation strategy.
  */
-public class ModelsComposition implements IgniteModel<Vector, Double>, Exportable<ModelsCompositionFormat> {
+public class ModelsComposition implements IgniteModel<Vector, Double>, Exportable<ModelsCompositionFormat>,
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/composition/ModelsComposition.java]: Maybe we should make this class final too?

File: modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java [hunk 2]
```diff
@@ -125,4 +128,9 @@ public DistanceMeasure distanceMeasure() {
             .addField("centroids", centersList)
             .toString();
     }
+
+    /** {@inheritDoc} */
+    @Override public List<Object> getDependencies() {
```
Comment (zaleslaw) [inline modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java]: We need to start the something like "Template Trainer Design Document" based on parent class to describe the purpose of all these methods and hidden limitations of implementation like this method, what do you think, @avplatonov ?

File: modules/ml/src/main/java/org/apache/ignite/ml/composition/ModelsComposition.java (hunk 1 above)
Comment (avplatonov) [reply modules/ml/src/main/java/org/apache/ignite/ml/composition/ModelsComposition.java]: I'm not sure, there is a child of this class (GDBModel) and I don't want to change an inheritance schema in this PR. Maybe we should create another ticker in Jira with a review of hierarchies of models (of course without changing APIs)

File: modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java (hunk 2 above)
Comment (avplatonov) [reply modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java]: What do you mean? Is this a kind of example class with a lot of comments or is it a document in wiki?

**Comment by avplatonov** [comment]
@zaleslaw I fixed conflicts. Please, have a look.

File: modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java (hunk 2 above)
Comment (zaleslaw) [reply modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java]: I suggest the page for Wiki in AI Wiki (you could start from the draft in the Google Doc and after review merge it into Wiki)

Comment (avplatonov) [reply modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansModel.java]: Ok! I'll do it. Unfortunately, I won't have free time the next week but I'll try to prepare some draft.

**Pull request merged**